e2e-tests/tests/headlamp.spec.ts

```diff
@@ -16,3 +16,16 @@ test("headlamp is there and so is minikube", async ({ page }) => {
 
   await expect(page).toHaveURL(/.*main/);
 });
+
+test('GET /plugins/list returns plugins list', async ({ page }) => {
+  const response = await page.goto('/plugins/list');
+  expect(response).toBeTruthy();
+
+  const text = await response.text();
```

Comment:
This could be redone using some notes from the API testing guide? https://playwright.dev/docs/api-testing

Something like (untested):
```ts
  const plugins= await request.get(''/plugins/list'');
  expect(await plugins.json().length).toBeGreaterThan(0)
```

Reply:
thing is plugins.json() is returning error. I think it’s not sending the response in incorrect json format. That why I had to use text and check

Reply:
Seems you found a bug. I guess it's maybe not returning the correct headers or something.

Reply:
Also the value is like this.

```sh
"[\"plugins/app-menus/main.js\",\"plugins/change-logo/main.js\",\"plugins/cluster-chooser/main.js\",\"plugins/details-view/main.js\",\"plugins/dynamic-clusters/main.js\",\"plugins/pod-counter/main.js\",\"plugins/sidebar/main.js\"]
```

Notice there is no `"` in the end.

Reply:
So I checked it's not a bug. I was not building latest backend code for my headlamp. It is indeed returning. `application/json` in headers. And API endpoint is changed to `/plugins` and how the response is sent is also changes. So refactoring the code accordingly